### DISCUSSION TITLE
The fastest console.log() ever?

### DISCUSSION THREAD

**Discussion by charnould** [oven-sh/bun#3616] [Ideas]
First, it might be a totally stupid idea.  
If so, say it and close this discussion :-)

**Foreword:** `console.log()` is here server-side.

**Idea**
1. Best practices say "do not `console.log()` in production"
2. But it's needed to debug your app, so some of us use `Pino` (or alternatives)
3. What if `Bun` extends `console.log()` to make it the fastest `Pino` ever/possible?

**Advantages:**
- `Bun` = battery-included
- Freaking fast from A to Z.

**Comment by simylein** [comment]
This would be awesome. I tried console.log(), console.write(), process.stdout.write() and they all seem fast enough, but maybe it is possible to go faster?

**Comment by vjpr** [comment]
FYI: `node:diagnostics_channel` should now be used for APM.

https://bun.sh/blog/bun-v0.7.2#node-js-node-diagnostics-channel-support

**Reply by charnould** [reply]
I did't know about `node:diagnostics_channel`.

Does it mean that for writing logs in production, the way to go is:
- use `node:diagnostics_channel`
- write something like below to log it (using `console.log`, `pino`, or whatever)?

```
channel.subscribe({ error }) => {
  console.error('An error', error)
})
```
